scripts/run_macos.sh

```diff
@@ -173,12 +208,17 @@ python_version="3.10"
 
 check_path_for_spaces
 
-print_highlight "Setting up Miniconda"
-install_miniconda
+read -p "Do you want to install Miniconda (recommended) y/n: " install_conda_choice
+# Convert user input to lowercase
+if [[ "$install_conda_choice" == "y" || "$install_conda_choice" == "Y" ]]; then
+    install_miniconda
+    print_highlight "Creating conda environment"
+    create_conda_env "$python_version"
+    activate_conda_env
+else
+    echo "miniconda installation skipped."
+fi
```

Comment:
We initially created scripts/run_* as installers for non-dev end users. They may not be familiar with Python and conda, and to avoid permission issues, we managed conda ourselves without caring if it already existed system-wide. This way, we didn't require user confirmation to install it. Additionally, miniconda is relatively small, so it didn't add much overhead to the application.

Reply:
🤔 🤔 🤔 For sure and I understand the want to support non-dev end users! Do you guys want to make installing miniconda a requirement and not optional? 

This is outside the scope of this PR, but I feel that if you want a non-dev end users to use this application possibly including a clickable `.app`, `.exe` package would be great - Maybe [pyinstaller](https://pyinstaller.org/en/v6.10.0/index.html) could be a good option for that. From my experience (and feel free to disagree with me), non technical users don't want to open up the terminal and type anything in :) it can be very scary.

Reply:
Just wanted to chime in as using conda was partly my decision. The motivation for making miniconda a requirement and not optional was to make the app as self-contained as possible. I wanted to avoid issues where the user's miniconda stopped working or got uninstalled making the app suddenly couldn't start anymore. Yes it would take more space but at the time I deem it as it's worth the tradeoff. I took inspiration from `text-generation-webui` (which also maintains its own conda binary) and `pre-commit` (which maintains its own python env to execute hooks).

> This is outside the scope of this PR, but I feel that if you want a non-dev end users to use this application possibly including a clickable .app, .exe package would be great - Maybe [pyinstaller](https://pyinstaller.org/en/v6.10.0/index.html) could be a good option for that. From my experience (and feel free to disagree with me), non technical users don't want to open up the terminal and type anything in :) it can be very scary.

Yes it would be ideal to have installation files such as `.app` and `.exe`. I love to have it available but at the time we simply didn't have the bandwidth to support it. And having all-in-one scripts was the next best option as it's easier for us to implement and maintain. And I completely agree with you that opening a terminal for a non-dev folks would be an alien experience, so we tried our best to mitigate that:
- the window `.bat` file was already double-click only
- for mac, I included a step-by-step to enable double-clicking the script in [the doc](https://cinnamon.github.io/kotaemon/#installation). I'm not familiar with mac os so this was the best way I could find, please suggest better alternatives if you know any.
- for linux, well, I just hoped that if the user is using Linux he/she would be comfortable with running commands in a terminal 😄

And tbh, we made that months ago and didn't expect it to get this many users. If enough non-dev users are interested in this project I hope we will have more helping hands to develop such clickable installation packages. (back then we even struggled to find a mac to test the scripts 😅) Now, whether to have that in the roadmap is entirely up to the current team. If you feel that it should be prioritized, feel free to create a feature request, or even better, contribute a PR, I'm sure it will be highly appreciated 😊

Reply:
Sounds good and totally reasonable. I'll change this so it doesn't ask for now and instead just installs it

You can change that if you'd like to have a clickable file for non-dev users on mac :)